frontend/src/components/CheckboxGroup/CheckboxGroupValidator.tsx

```diff
@@ -0,0 +1,221 @@
+import { List } from 'immutable';
+import React, { FunctionComponent, useState } from 'react';
+import { Filters, OperationStepMap } from '../../types/operations';
+import { BasicModal } from '../BasicModal';
+
+interface ComponentProps {
+  steps?: List<OperationStepMap>;
+}
+
+const CheckboxGroupValidator: FunctionComponent<ComponentProps> = (props) => {
+  const [showModal, setShowModal] = useState<boolean>(false);
+  const [modalMessage, setModalMessage] = useState<string>('');
```

Comment:
I think the modalMessage state can be enough on it's own ... Unless u have a situation where it happens, if there's nothing to show, hide the modal.

Reply:
Updated